external/header.php

```diff
@@ -51,7 +51,28 @@ public static function __callstatic($name, $arguments)
 }
 
 // Only users from authorized IP addresses may control Profiling
-if ($controlIPs === false || in_array($_SERVER['REMOTE_ADDR'], $controlIPs) || PHP_SAPI == 'cli')
+function get_client_ip() {
+    $ipaddress = '';
```

Comment:
1. Are all these headers really need to be checked?
2. Blindly trusting IP provided in these headers actually open a security hole, where an attacker can put needed IP in a header and force website to run slower due fact, that xhprof is collecting analytics data. And in fact it will print profiling link back to attacker, which in turn would reveal some sensitive info about website itself.